src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java

```diff
@@ -86,4 +91,14 @@ public void deleteDeck(Long deckId) {
         deckRepository.deleteDeckById(deckId);
     }
 
+    public void createNewDeck(Deck newDeck, Long category_id) throws NotAuthorisedUserException {
+        User user = userService.getAuthorizedUser();
+        Deck deck = new Deck();
+        deck.setName(newDeck.getName());
+        deck.setDescription(newDeck.getDescription());
+        deck.setCategory(categoryRepository.findById(category_id));
+        deck.setDeckOwner(user);
+        Deck save = deckRepository.save(deck);
+        newDeck.setId(save.getId());
```

Comment:
Why can't you just use newDeck instead of creating a new object? I don't say it's wrong, just asking.